### ISSUE TITLE
训练显存增长，直至崩溃的问题？

### ISSUE THREAD

**Issue by EkilGitHub** [AllentDan/LibtorchDetection#1]
你好，用您的代码训练自己的样本过程中，存在显存不断增长，直至程序崩溃的问题？
请问，可能是什么原因导致的。

**Comment by AllentDan** [comment]
Hi, 如果你不着急的话我周末会检查下代码。看看是否有显存泄漏问题，着急的话可以帮我定位下问题然后提个PR上来😆

**Comment by EkilGitHub** [comment]
> Hi, 如果你不着急的话我周末会检查下代码。看看是否有显存泄漏问题，着急的话可以帮我定位下问题然后提个PR上来😆

好的，谢谢回复。昨天我对比了一下Detector.cpp中train部分的代码，和python版本的区别，并未发现不同。而尝试loss.detach()也并未起到作用。或许不是Detector::Train这个地方 显存泄露了。

**Comment by EkilGitHub** [comment]
> Hi, 如果你不着急的话我周末会检查下代码。看看是否有显存泄漏问题，着急的话可以帮我定位下问题然后提个PR上来😆
抱歉，今天测试了一下，发现不是代码存在显存泄露的问题，而是我这边存在资源不足。

您好，因为我们导师是做嵌入式方面深度学习的，所以十分感谢提供yolo/libtorch的代码来进行学习。由于刚接触这部分知识，有些疑惑，如果您有时间的话，希望能够帮忙简单答疑。
问题描述：之前训练集样本量少的时候，并未发现显存不足的现象，但此次增大样本量后，就出现问题了。在0、1、2epoch时，显存并未明显增加，但当epoch = 3时，显存增加明显。可以定位到是如下这部分代码起了作用。
``
if (epoc_count < int(num_epochs / 10)) {
			for (auto mm : pretrained_dict)
			{
				if (strstr(mm.key().c_str(), "yolo_head"))
				{
					mm.value().set_requires_grad(true);
				}
				else
				{
					mm.value().set_requires_grad(false);
				}
			}
		}
		else {
			for (auto mm : pretrained_dict) {
				mm.value().set_requires_grad(true);
			}
		}
``
但对应这段代码有两个我不懂的地方。
1、为什么在epoch < 3 的时候，设置requires_grad(true)，而 ≥ 3 的时候，全部设置为true。
2、这样设置requires_grad的目的是什么。
2、为什么增大训练集后，训练中，显存占用会增大。
